ouroboros-consensus/tools/db-convert/Main.hs

```diff
@@ -0,0 +1,275 @@
+            }
+          protocolMagicId = CB.unAnnotated . getAProtocolMagicId $ CC.Genesis.configProtocolMagic cfg
+          epochSlots = CC.Genesis.configEpochSlots cfg
+          securityParam = SecurityParam $ CC.unBlockCount k
+          k = CC.Genesis.configK cfg
+          args registry =
+            (ChainDB.defaultArgs @blk (toFilePath dbDir))
+              { ChainDB.cdbGenesis = return initLedgerState
+              , ChainDB.cdbDecodeBlock = Byron.decodeByronBlock epochSlots
+              , ChainDB.cdbDecodeChainState = Byron.decodeByronChainState
+              , ChainDB.cdbDecodeHash = Byron.decodeByronHeaderHash
+              , ChainDB.cdbDecodeLedger = Byron.decodeByronLedgerState
+              , ChainDB.cdbEncodeBlock = Byron.encodeByronBlock
+              , ChainDB.cdbEncodeChainState = Byron.encodeByronChainState
+              , ChainDB.cdbEncodeHash = Byron.encodeByronHeaderHash
+              , ChainDB.cdbEncodeLedger = Byron.encodeByronLedgerState
+              , -- Policy
+              ChainDB.cdbValidation = ImmDB.ValidateAllEpochs
+              , ChainDB.cdbBlocksPerFile = 10
+              , ChainDB.cdbMemPolicy = LgrDB.defaultMemPolicy securityParam
+              , ChainDB.cdbDiskPolicy = LgrDB.defaultDiskPolicy securityParam 20000
+              , -- Integration
+              ChainDB.cdbNodeConfig = WithEBBNodeConfig $ EncNodeConfig
+                { encNodeConfigP = PBftNodeConfig
+                    { pbftParams = PBftParams
+                        { pbftSecurityParam = securityParam
+                        , pbftNumNodes = 7
+                        , pbftSignatureWindow = coerce k
+                        , pbftSignatureThreshold = 0.22
+                        }
+                    , pbftIsLeader = Nothing
+                    }
+                , encNodeConfigExt = ByronConfig
+                  { pbftProtocolMagic = CC.Genesis.configProtocolMagic cfg
+                  , pbftProtocolVersion = CC.Update.ProtocolVersion 1 0 0
+                  , pbftSoftwareVersion = CC.Update.SoftwareVersion
+                    (CC.Update.ApplicationName "Cardano SL")
+                    2
+                  , pbftGenesisConfig = cfg
+                  , pbftGenesisHash = CC.Genesis.configGenesisHash cfg
+                  , pbftEpochSlots = epochSlots
+                    -- These aren't needed here, because this part of the code will never be exercised.
+                  , pbftSecrets = error "Attempt to use PBFT secrets in DB validation!"
+                  }
+                }
+              , ChainDB.cdbEpochSize = const (return . EpochSize . unEpochSlots $ epochSlots)
+              , ChainDB.cdbIsEBB = \blk -> case Byron.unByronBlockOrEBB blk of
+                  CC.ABOBBlock _    -> Nothing
+                  CC.ABOBBoundary ebb -> Just (CC.boundaryHashAnnotated ebb)
+              , -- Misc
+              ChainDB.cdbTracer = if verbose
+              then
```

Comment:
Please indent `then` and `else` (with two spaces) to make it clear that this code is part of the RHS of`ChainDB.cdbTracer`